google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ConnectionStatementExecutorTest.java

```diff
@@ -237,4 +242,102 @@ public void testStatementSetPgTransactionModeNoOp() {
     verify(connection, never()).setTransactionMode(TransactionMode.READ_ONLY_TRANSACTION);
     verify(connection, never()).setTransactionMode(TransactionMode.READ_WRITE_TRANSACTION);
   }
+
+  private void assertExplain(ResultSet rs){
+    int countRows = 0;
+    while (rs.next()) {
+      ++countRows;
+    }
+
+    Assert.assertEquals(1, countRows);
+    Assert.assertEquals(1, rs.getColumnCount());
+    Assert.assertEquals(1, rs.getColumnIndex("query plan"));
+    Assert.assertEquals(0, rs.getColumnIndex("random column"));
+
+  }
+
+  private void assertExplainAnalyse(ResultSet rs){
+    int countRows = 0;
+    while (rs.next()) {
+      ++countRows;
+    }
+
+    Assert.assertEquals(3, countRows);
+    Assert.assertEquals(1, rs.getColumnCount());
+    Assert.assertEquals(1, rs.getColumnIndex("query plan"));
+    Assert.assertEquals(0, rs.getColumnIndex("random column"));
+
+
+  }
+
+  @Test
+  public void testStatementExplain() {
+
+    ExplainCommandConverter converter = new ExplainCommandConverter();
+
+    /*Test for valid EXPLAIN*/
+    String sql = converter.convert("explain select * from table");
+
+    ResultSet expectedRs = mock(ResultSet.class);
+    when(expectedRs.next()).thenReturn(true, false);
+    when(expectedRs.getColumnCount()).thenReturn(1);
+    when(expectedRs.getColumnIndex("query plan")).thenReturn(1);
+
+    when(connection.analyzeQuery(any(), any())).thenReturn(expectedRs);
+
+    ResultSet rs = subject.statementExplain(sql).getResultSet();
+
+    assertExplain(rs);
+    /*Test for EXPLAIN ANALYSE*/
```

Comment:
Split different test cases into different methods. That makes it easier to detect what fails if a test fails.